examples/examples/instrumented_error.rs

```diff
@@ -0,0 +1,92 @@
+//! This example demonstrates using the `tracing-error` crate's `SpanTrace` type
+//! to attach a trace context to a custom error type.
+#![deny(rust_2018_idioms)]
+use std::error::Error;
+use std::fmt;
+use tracing_error::{prelude::*, ErrorLayer};
+use tracing_subscriber::{prelude::*, registry::Registry};
+
+#[derive(Debug)]
+struct FooError {
+    message: &'static str,
+}
+
+// Arbitrary user defined error type for demonstration purposes only
+impl FooError {
+    fn new(message: &'static str) -> Self {
+        Self { message }
+    }
+}
+
+impl Error for FooError {}
+
+impl fmt::Display for FooError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.pad(self.message)
+    }
+}
+
+#[tracing::instrument]
+fn do_something(foo: &str) -> Result<&'static str, impl Error + Send + Sync + 'static> {
+    // Results can be instrumented with a `SpanTrace` via the `InstrumentResult` trait
+    do_another_thing(42, false).in_current_span()
+}
+
+#[tracing::instrument]
+fn do_another_thing(
+    answer: usize,
+    will_succeed: bool,
+) -> Result<&'static str, impl Error + Send + Sync + 'static> {
+    // Errors can also be instrumented directly via the `InstrumentError` trait
+    Err(FooError::new("something broke, lol").in_current_span())
+}
+
+
+#[tracing::instrument]
+fn main() {
+    let subscriber = Registry::default()
+        .with(tracing_subscriber::fmt::Layer::default())
+        // The `ErrorLayer` subscriber layer enables the use of `SpanTrace`.
+        .with(ErrorLayer::default());
+    tracing::subscriber::set_global_default(subscriber).expect("Could not set global default");
+    match do_something("hello world") {
+        Ok(result) => println!("did something successfully: {}", result),
+        Err(e) => {
+            eprintln!("printing error chain naively");
+            print_naive_spantraces(&e);
+
+            eprintln!("\nprinting error with extract method");
+            print_extracted_spantraces(&e);
+        }
+    };
+}
+
+// Iterate through the source errors and check if each error is actually the attached `SpanTrace`
+// before printing it
+fn print_extracted_spantraces(error: &(dyn Error + 'static)) {
+    let mut error = Some(error);
+    let mut ind = 0;
+
+    while let Some(err) = error {
+        if let Some(spantrace) = err.span_trace() {
+            eprintln!("Span Backtrace:\n{}", spantrace);
```

Comment:
Doesn't the span trace's `fmt::Display` impl already print "span backtrace:"?

Reply:
No, only the TracedError display impl prints the `span backtrace:` header, once you extract it you're back to printing the inner span_trace sans header.